Environment_Engine/Query/IsContaining.cs

```diff
@@ -168,16 +235,40 @@ public static bool IsContaining(this List<Panel> panels, Point point, bool accep
             return isContained; //If the number of intersections is odd the point is outsde the space
         }
 
-        [Description("Defines whether an Environment Space contains a provided point.")]
-        [Input("space", "An Environment Space object defining a perimeter to build a 3D volume from and check if the volume contains the provided point.")]
-        [Input("spaceHeight", "The height of the space.", typeof(BH.oM.Quantities.Attributes.Length))]
-        [Input("points", "The points being checked to see if it is contained within the bounds of the 3D volume.")]
-        [Input("acceptOnEdges", "Decide whether to allow the point to sit on the edge of the space, default false.")]
-        [Output("isContaining", "True if the point is contained within the space, false if it is not.")]
-        public static List<bool> IsContaining(this Space space, double spaceHeight, List<Point> points, bool acceptOnEdges = false)
+        [Description("Defines whether a collection of Environment Panels contains each of a provided list of list of points.")]
+        [Input("panels", "A collection of Environment Panels to check with.")]
+        [Input("pointLists", "The List of Lists of points to check to see if each List of points are contained within the bounds of the panels.")]
+        [Input("acceptOnEdges", "Decide whether to allow the points to sit on the edge of the panel, default false.")]
+        [Input("acceptPartialContainment", "Decide whether to allow some of the points to sit outside the panels as long as at least one is within them.")]
+        [Output("isContaining", "True if the points of each sublist are contained within the bounds of the panels, false if it is not for each sublist of points provided.")]
+        public static List<bool> IsContaining(this List<Panel> panels, List<List<Point>> pointLists, bool acceptOnEdges = false, bool acceptPartialContainment = false, double tolerance = BH.oM.Geometry.Tolerance.Distance)
```

Comment:
This method would require versioning, but also shouldn't be changed like this, this will change how other downstream and existing workflows operate by changing from `List<Point>` to `List<List<Point>>`. Let's discuss the necessity in an automation labs catch up please @enarhi

Reply:
Resolved as per our discussion @FraserGreenroyd , method is a new method and is private, so no concerns per the above.